File: CS/WinSample.Module.Win/CheckableSimpleAction.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using System.ComponentModel;

namespace WinSample.Module.Win {
    public class CheckableSimpleAction : SimpleAction {
        public CheckableSimpleAction(IContainer container) : base(container) { }

        private bool _Checked = false;
        public bool Checked {
            get {
                return _Checked;
            }
            set {
                if (_Checked != value) {
                    _Checked = value;
                    DoCheckedStateChanged();
                    if (Enabled.ResultValue && Active.ResultValue) {
                        DoExecute();
                    }
                }
            }
        }

        private void DoCheckedStateChanged() {
            if (CheckedStateChanged != null) {
                CheckedStateChanged(this, new EventArgs());
            }
        }

        public event EventHandler CheckedStateChanged;
    }
}

File: VB/WinSample.Module.Win/CheckableSimpleAction.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports System.ComponentModel

Namespace WinSample.Module.Win
	Public Class CheckableSimpleAction
		Inherits SimpleAction
		Public Sub New(ByVal container As IContainer)
			MyBase.New(container)
		End Sub

		Private _Checked As Boolean = False
		Public Property Checked() As Boolean
			Get
				Return _Checked
			End Get
			Set(ByVal value As Boolean)
				If _Checked <> value Then
					_Checked = value
					DoCheckedStateChanged()
					If Enabled.ResultValue AndAlso Active.ResultValue Then
						DoExecute()
					End If
				End If
			End Set
		End Property

		Private Sub DoCheckedStateChanged()
			RaiseEvent CheckedStateChanged(Me, New EventArgs())
		End Sub

		Public Event CheckedStateChanged As EventHandler
	End Class
End Namespace
